play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSClient.scala

```diff
@@ -79,8 +79,12 @@ class StandaloneAhcWSClient @Inject() (asyncHttpClient: AsyncHttpClient)(implici
   private def validate(url: String): Unit = {
     // Recover from https://github.com/AsyncHttpClient/async-http-client/issues/1149
     Try(Uri.create(url)).transform(Success(_), {
+      case iae: IllegalArgumentException =>
+        Failure(new IllegalArgumentException(s"Invalid URL $url", iae))
       case npe: NullPointerException =>
         Failure(new IllegalArgumentException(s"Invalid URL $url", npe))
+      case e: Throwable =>
+        Failure(e)
     }).get
```

Comment:
`Try { ... }.transform(Success(_), _), { ... }).get` is really just a try-catch block…

Reply:
Good point. Updated.